presto-parser/src/main/java/com/facebook/presto/sql/tree/GenericLiteral.java

```diff
@@ -39,6 +41,13 @@ private GenericLiteral(Optional<NodeLocation> location, String type, String valu
         super(location);
         requireNonNull(type, "type is null");
         requireNonNull(value, "value is null");
+        if (type.equalsIgnoreCase("X")) {
+            // we explicitly disallow "X" as type name, so if the user arrived here,
+            // it must be because that he intended to give a binaryLiteral instead, but
+            // added whitespace between the X and quote
+            throw new ParsingException("Did you intend to write a Binary Literal instead? 'X' is not allowed to be the type name,"
+                            + ", seen [" + type + "]");
```

Comment:
Remove the `seen ...` part. Pass the location as an argument to the constructor.